src/internal/metrics.ts

```diff
@@ -102,6 +128,37 @@ export const Metrics = {
     }
   },
 
+  /**
+   * Calls Console Platform's client v2 logging JS API with provided metric name and detail.
+   * Does nothing if Console Platform client logging JS is not present in page.
+   */
+  sendMetricV2(metricName: string, metric: MetricsV2EventItem): void {
```

Comment:
Thoughts on appropriate error handling for this? Wrapping the entire function in a try catch to silently fail as a catch all scenario vs letting the errors bubble up.

Reply:
Depends on guarantees from receiving end. `AWSC.Clog.log` API is asynchronous inside, so it always succeeded (because it only put message in a queue, sent with intervals).

If their sync logic is also rather small and they do not throw any validation errors, we can keep it without try/catch. 

But this is only true if we replace `window.panorama` with something else. We do not want our components crash via `window.panorama = {}`, throwing `TypeError: panorama is not a function`

Reply:
P.S. another comment to the same line – can we call it `sendPanoramaMetric`? Nouns are easier to recall when reading unfamiliar code, than version numbers

Reply:
Event detail does not contain any user input (and never should, you can ask me in DM why). If stringily fails, it is a bug on our end, should be captured with tests